[johngao122] iP

johngao122 opened the pull request:
# Quip - Your Quirky Task Management Companion 🎯

A Java-based task management application that helps you keep track of your tasks with a dash of humor!

## Features
* Manages three types of tasks:
  * Simple todos
  * Deadlines with due dates
  * Events with start and end times
* Persistent storage in CSV format
* Case-insensitive command parsing
* Date-based task filtering

## Getting Started
1. Ensure you have Java 11 or above installed
2. Clone this repository
3. Compile the project
4. Run `Quip.java`

## Sample Usage
```java
// Create a new todo
todo Read a book

// Add a deadline
deadline Return library book /by 2024-02-15 14:30

// Schedule an event
event Team meeting /from 2024-02-10 09:00 /to 2024-02-10 10:30
```

## Project Status
- [x] Basic task management
- [x] Data persistence
- [x] Command parsing
- [ ] Natural language processing
- [ ] GUI implementation

## Design Philosophy
> "A task manager should be your friend, not your boss."

This project follows the **Command Pattern** for handling user inputs and uses *Object-Oriented Programming* principles throughout its implementation. It's built with ~~complexity~~ simplicity in mind.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

johngao122 committed:
level-0

johngao122 committed:
level-0

johngao122 committed:
level 1

johngao122 committed:
level 2

johngao122 committed:
level 3

johngao122 committed:
level 4

johngao122 committed:
TextUiTesting

johngao122 committed:
Level 5

johngao122 committed:
Level 6

johngao122 committed:
A-enums

johngao122 committed:
Changed variable names to match specified coding style requirements

johngao122 committed:
Implement level 7: storage

    Add Storage class to save/load tasks to/from CSV file(s) with format:
    T,true,read book
    D,false,return book,June 6th
    E,false,2023-11-20 00:00,2023-11-20 12:00

    - Add load method with proper error handling
    - Add save method that stores tasks in CSV format
    - Handle missing directories and files
    - Add unit tests for storage operations

johngao122 committed:
Merge branch-Level-7 into master

johngao122 committed:
Integrate datetime into deadline

johngao122 committed:
Integrate datetime into event

johngao122 committed:
Add listTasksonDate method in Quip

johngao122 committed:
Merge branch-Level-8 into master

johngao122 committed:
refactor: implement OOP structure with command pattern

Extract core functionality into dedicated classes (Ui, Storage, Parser, TaskList)
Implement Command pattern with abstract Command class
Add specialized command classes for task operations
Migrate task management logic into TaskList class
Improve exception handling and user feedback

johngao122 committed:
Merge branch-A-MoreOOP into master

johngao122 committed:
refactor: organize classes into packages for better code organization

- Create package structure under src/main/java/quip
- Group related classes into command, exception, parser, storage, task, ui packages
- Update imports to reference new package paths

johngao122 committed:
Merge branch-A-Packages into master

johngao122 committed:
Add gradle support

johngao122 committed:
Add gradle support for quip

johngao122 committed:
Merge branch-A-Gradle into master

johngao122 committed:
Add Junit test for command, parser,storage,task

johngao122 committed:
Merge branch-A-JUnit

johngao122 committed:
Add JAR support

johngao122 committed:
Merge branch-A-Jar

johngao122 committed:
Add JavaDocs for Quip,command,parser,tasklist

johngao122 committed:
Adjust constant names for Storage.java

johngao122 committed:
Add FindCommand and showMatchingTasks to implement find task command

johngao122 committed:
Merge branch-A-JavaDoc

johngao122 committed:
Merge branch-A-CodingStandard

johngao122 committed:
Merge branch-Level-9

johngao122 committed:
Parser: Convert to proper utility class

Make Parser a utility class by:
* Add private constructor to prevent instantiation
* Mark class as final
* Remove wildcard import
* Remove unnecessary default constructor

johngao122 committed:
Storage: Fix naming conventions and add documentation

* Rename FILE_PATH and FILE_NAME to follow Java naming convention
* Add class-level Javadoc
* Add method-level Javadoc for public methods

johngao122 committed:
Add checkstyle

johngao122 committed:
Merge branch-A-CheckStyle

johngao122 committed:
Add JavaFX chat interface

The console interface is replaced with a modern chat GUI using JavaFX.

* Create DialogBox,UserDialogBox,QuipDialogBox,Main and Launcher class for the chat interface
* Modify Quip class to use JavaFX
* Add CSS styling

johngao122 committed:
Add JavaFX UI with commands

* Create JavaFxUi class that extends base Ui class
* Override all output methods to use QuipDialogBox
* Update Main class to use JavaFxUi implementation
* Improve code organization in Main class

This change allows users to see all interactions within the GUI interface

johngao122 committed:
merge branch-Level-10 to master

johngao122 committed:
Add addTasks in TaskList

johngao122 committed:
merge branch-A-Varargs

sherwinee commented on src/main/java/quip/Main.java:
```diff
@@ -0,0 +1,108 @@
+import javafx.scene.layout.VBox;
+import javafx.scene.layout.HBox;
+import javafx.stage.Stage;
+import quip.command.Command;
```
I like your organisation of your multiple import statements! Perhaps we can add an empty line here?

sherwinee commented on src/main/java/quip/ui/Ui.java:
```diff
@@ -0,0 +1,117 @@
+    }
+
+
+    public void showLoadingError() {
```
I like how you used a verb for actions in your method name here

sherwinee commented on src/main/java/quip/task/Task.java:
```diff
@@ -0,0 +1,43 @@
+    public Task(String task, TaskType type) {
+        this.task = task;
+        this.type = type;
+        this.isDone = false;
```
I like how your boolean name sounds like a boolean

sherwinee commented on src/main/java/quip/ui/Ui.java:
```diff
@@ -0,0 +1,117 @@
+
+public class Ui {
+    private static final String LINE = "_____________________________________________________________";
+    private static final String NAME = "Quip";
```
I like your use of a noun for naming your variable and using SCREAMING_SNAKE_CASE for them since they are constants

sherwinee reviewed (commented):
LGTM! Honestly I dont have much to say as your code seems to follow naming conventions well as mentioned in week 4 contents. I've highlighted some key good usages.

botosek commented on src/main/java/quip/command/Command.java:
```diff
@@ -0,0 +1,16 @@
+import quip.task.TaskList;
+import quip.ui.Ui;
+
+public abstract class Command {
```
I like how you separated each command as an individual class on its own. I feel that it makes organizing your code alot easier!

botosek commented on src/main/java/quip/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    private static final DateTimeFormatter DISPLAY_FORMATTER =
```
I like that you separate your formatter constants for both your display and files. I feel that this might make it easier to debug too.

botosek commented on src/main/java/quip/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+public class Deadline extends Task {
+    private static final DateTimeFormatter DISPLAY_FORMATTER =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd h:mm a");
+    private static final DateTimeFormatter FILE_FORMATTER =
```
I like the use of caps for your constant names.

botosek commented on src/main/java/quip/command/FindCommand.java:
```diff
@@ -0,0 +1,28 @@
+package quip.command;
+
+import quip.exception.QuipException;
+import quip.storage.Storage;
+import quip.task.Task;
+import quip.task.TaskList;
+import quip.ui.Ui;
+
+import java.util.List;
```
I like how the order of your import statements are standardized across your files.

johngao122 committed:
Generate executable JAR with dependencies

- Use shadowJar plugin to create the executable
- Bundle JavaFX modules into the JAR
- Configure main class as quip.Launcher

johngao122 committed:
Add assertions to document code assumptions

johngao122 committed:
Improve code quality

johngao122 committed:
Refactor TaskList to use Java Streams

johngao122 committed:
Merge pull request #1 from johngao122/branch-A-Assertions

Add assertions to document code assumptions

johngao122 committed:
Merge pull request #2 from johngao122/branch-A-CodeQuality

Improve code quality

johngao122 committed:
Merge pull request #3 from johngao122/branch-A-Streams

Refactor TaskList to use Java Streams

johngao122 committed:
Set up continuous integration workflow

johngao122 committed:
Add remind command to Quip

Use "remind" command to get notified of any deadlines or events that are due within 24 hours

johngao122 committed:
merge branch-B-Reminders

johngao122 committed:
Add duplicate task detection

There are several methods that convert a collection to a set. In some
cases the conversion is in-lined as a code block in another method.

johngao122 committed:
merge branch-C-DetectDuplicates

johngao122 committed:
Add error message highlighting to chat interface

johngao122 committed:
merge branch-A-BetterGui

johngao122 committed:
Tweak Personality of Quip

johngao122 committed:
merge branch-A-Personality

johngao122 committed:
Fix coding standard for some parts

johngao122 committed:
Add user guide

johngao122 committed:
Adjusted CI workflow

johngao122 committed:
Adjust CI workflow

johngao122 committed:
Fix checkstyle errors

johngao122 committed:
Fix locale errors between github CI and local

johngao122 committed:
Fix locale error and checkstyle error for tests

johngao122 committed:
Prepare gradle configuration for release

johngao122 committed:
Fix gradle wrapper error for release

johngao122 committed:
Fix build.gradle to create shadowJar for windows
